src/Models/Product.php

```diff
@@ -85,6 +89,59 @@ public function getCasts(): array
         return $this->casts;
     }
 
+    // TODO: Maybe move all these search related
+    // methods to a separated trait?
+    public function toSearchableArray(): array
+    {
+        $product = $this->toArray();
+
+        // dd($product);
+
+        // TODO: Maybe double check / handle what attributes are
+        // getting indexed here. From makeAllSearchableUsing
+        // we're getting the correct data, but when you
+        // index just one; I don't think that query
+        // will be used resulting in all attrs.
+
+        // Customize the data array...
+        // cast to the correct types! (int), etc
```

Comment:
Casting to the correct types should already be handled by the $casts right?

I do agree that it might be a good idea to pass the data through an eventy filter(?) so it can be changed as desired.